### ISSUE TITLE
kserve v0.11.2 can't be poetry installed

### ISSUE THREAD

**Issue by timothyjlaurent** [kserve/kserve#3300]
/kind bug

**What steps did you take and what happened:**

I'm having difficulty installing kserve with poetry:
```
❯ poetry add kserve
Configuration file exists at /Users/me/Library/Preferences/pypoetry, reusing this directory.

Consider moving TOML configuration files to /Users/me/Library/Application Support/pypoetry, as support for the legacy directory will be removed in an upcoming release.
Using version ^0.11.2 for kserve

Updating dependencies
Resolving dependencies... (3.3s)

Package operations: 19 installs, 2 updates, 0 removals

  • Downgrading protobuf (4.25.1 -> 3.20.3)
  • Downgrading grpcio (1.51.3 -> 1.49.1)
  • Installing attrs (23.1.0)
  • Installing frozenlist (1.4.0)
  • Installing multidict (6.0.4)
  • Installing sniffio (1.3.0)
  • Installing aiosignal (1.3.1)
  • Installing anyio (4.1.0)
  • Installing yarl (1.9.4)
  • Installing aiohttp (3.9.1)
  • Installing h11 (0.14.0)
  • Installing starlette (0.27.0)
  • Installing fastapi (0.95.2)
  • Installing prometheus-client (0.13.1)
  • Installing python-dotenv (1.0.0)
  • Installing python-rapidjson (1.13)
  • Installing uvicorn (0.19.0)
  • Installing uvloop (0.19.0)
  • Installing ray (2.4.0): Failed

  RuntimeError

  Unable to find installation candidates for ray (2.4.0)

  at ~/.local/pipx/venvs/poetry/lib/python3.11/site-packages/poetry/installation/chooser.py:73 in choose_for
       69│
       70│             links.append(link)
       71│
       72│         if not links:
    →  73│             raise RuntimeError(f"Unable to find installation candidates for {package}")
       74│
       75│         # Get the best link
       76│         chosen = max(links, key=lambda link: self._sort_key(package, link))
       77│

Cannot install ray.
```
~So what's confusing to me is that the ray dependency in kserve's pyproject.toml is 2.8.0 - so why is it looking for 2.4.0 ??~
This is because these changes haven't been deployed yet

**What did you expect to happen:**

~It should install Ray 2.8.0  since that is specified in the source code requirements.toml~

`poetry add kserve` should work without errors.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

I pulled the tarball from pypi and indeed the dependencies are outated:

```
ray = { version = "~2.4.0", extras = ["serve"] }
grpcio = "^1.49.1"
```

**Environment:**

- KServe Version: 11.2.0

- OS (e.g. from `/etc/os-release`): M2 mac

**Comment by timothyjlaurent** [comment]
I see what's goin on -- these updates haven't been released yet.

```
kserve = { git = "https://github.com/kserve/kserve.git", subdirectory = "python/kserve", rev = "c1c5efe" }
```

That is workaround until the changes get released

**Comment by timothyjlaurent** [comment]
So I guess this upgrade to ray hasn't made it into the 0.11.2 release, but I think this is still a problem since we can't install kserve with poetry

**Comment by sivanantha321** [comment]
Please use the patched 0.12.1 version
